File: music-cloud-server/src/main/java/com/kulygin/musiccloud/domain/Playlist.java
package com.kulygin.musiccloud.domain;

import lombok.*;

import javax.persistence.*;
import java.util.Set;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(schema = "public")
@EqualsAndHashCode(of = { "id"})
@ToString(exclude = {"tracks", "user"})
public class Playlist {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String name;

    @ManyToMany(fetch = FetchType.LAZY, mappedBy = "playlists")
    private Set<Track> tracks;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;
}

File: music-cloud-web/src/main/web/src/app/dto/playlist.ts
import { Track } from './track';

export class Playlist {
    private _id: number;
    private _name: string;
    private _tracks: Track[] = [];

    constructor(data: any) {
        this._id = data.id;
        this._name = data.name;
        data.tracks.forEach(track => {
            this._tracks.push(new Track(track));
        });
    }

    get id(): number {
        return this._id;
    }

    set id(value: number) {
        this._id = value;
    }

    get name(): string {
        return this._name;
    }

    set name(value: string) {
        this._name = value;
    }

    get tracks(): Track[] {
        return this._tracks;
    }

    set tracks(value: Track[]) {
        this._tracks = value;
    }
}
